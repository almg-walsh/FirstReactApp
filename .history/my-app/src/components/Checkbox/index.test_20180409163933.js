import React from "react";
import { mount } from "enzyme";
import Checkbox from "./index";

describe("Advert", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hide when button clicked", () => {
    const wrapper = mount(<Checkbox />);
    const event = { target: { isChecked: false } };
    wrapper.find("input").simulate("change", event);
    expect(wrapper.state().isChecked).toEqual(false);
  });
});
