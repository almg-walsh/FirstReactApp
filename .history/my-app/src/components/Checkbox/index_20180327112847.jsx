import React, { Component } from "react";

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: false
  }
  this.handleChange = this.handleChange.bind(this);
}

handleChange = () => {
  const { handleChange } = this.props;

  this.setState(({ isChecked }) => ({
    isChecked: !isChecked
  }));

  handleChange();
};

    render() {
      const { label } = this.props;
  
      return (
      <div className="checkbox">
          <label>
            <input
              type="checkbox"
              value={label}
              checked={this.props.isChecked}
              onChange={this.handleChange}
            />
            {label}
          </label>
        </div>
      );
    }
  }

  export default Checkbox;