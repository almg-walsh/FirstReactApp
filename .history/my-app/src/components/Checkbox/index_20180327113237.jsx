import React, { Component } from "react";

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: false
  }
  this.handleChange = this.handleChange.bind(this);
}

handleChange = () => {
  const { onClickChange } = this.props;

  this.setState(({ isChecked }) => ({
    isChecked: !isChecked
  }));

  onClickChange(this.state.isChecked);
};

    render() {
      const { label } = this.props;
  
      return (
      <div className="checkbox">
          <label>
            <input
              type="checkbox"
              value={label}
              checked={this.props.isChecked}
              onChange={this.handleChange}
            />
            {label}
          </label>
        </div>
      );
    }
  }

  export default Checkbox;