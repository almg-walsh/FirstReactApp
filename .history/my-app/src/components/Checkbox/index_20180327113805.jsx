import React, { Component } from "react";

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

    handleChange(e){
      
      this.setState({isChecked: false});
    }

    render() {
      const { label } = this.props;
  
      return (
      <div className="checkbox">
          <label>
            <input
              type="checkbox"
              value={label}
              checked={this.props.isChecked}
              onChange={this.handleChange}
            />
            {label}
          </label>
        </div>
      );
    }
  }

  export default Checkbox;