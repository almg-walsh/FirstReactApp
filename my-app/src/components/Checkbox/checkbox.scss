.checkboxes {
    margin: 0 auto;
  }