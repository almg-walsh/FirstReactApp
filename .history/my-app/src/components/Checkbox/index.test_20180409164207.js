import React from "react";
import { mount } from "enzyme";
import Checkbox from "./index";

describe("Advert", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should be not checked", () => {
    const wrapper = mount(<Checkbox />);
    const event = { target: { isChecked: false } };
    wrapper.find("input").simulate("change", event);
    expect(wrapper.state().isChecked).toEqual(false);
  });
});
