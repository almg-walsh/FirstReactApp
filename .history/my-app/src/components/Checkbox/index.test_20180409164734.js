import React from "react";
import { mount } from "enzyme";
import Checkbox from "./index";

describe("Advert", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not be checked", () => {
    const wrapper = mount(<Checkbox />);
    const event = { target: { isChecked: false } };
    wrapper.find("input").simulate("change", event);
    expect(wrapper.state().isChecked).toEqual(false);
    expect(wrapper.props().handleCheckboxChange(false));
  });
});
