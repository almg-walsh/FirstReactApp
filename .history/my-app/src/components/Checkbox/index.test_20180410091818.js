import React from "react";
import { mount } from "enzyme";
import Checkbox from "./index";

describe("Advert", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not be checked", () => {
    const wrapper = mount(<Checkbox />);
    const checkbox = wrapper.find("input");
    checkbox.simulate("change", { target: { checked: true } });
    expect(wrapper.state().isChecked).toEqual(true);
  });
});
