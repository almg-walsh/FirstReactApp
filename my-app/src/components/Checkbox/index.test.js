import React from "react";
import { mount } from "enzyme";
import Checkbox from "./index";

describe("Advert", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });
});
