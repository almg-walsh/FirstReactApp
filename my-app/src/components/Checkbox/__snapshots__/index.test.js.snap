// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advert should render correctly 1`] = `
<Checkbox>
  <div
    className="checkbox"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </label>
  </div>
</Checkbox>
`;
